src/core/device.c

```diff
@@ -878,7 +876,8 @@ static void device_enumerate(Manager *m) {
                 if (sd_device_get_syspath(dev, &sysfs) < 0)
                         continue;
 
-                device_update_found_by_sysfs(m, sysfs, DEVICE_FOUND_UDEV, DEVICE_FOUND_UDEV);
+                LIST_FOREACH_SAFE(same_sysfs, d, n, hashmap_get(m->devices_by_sysfs, sysfs))
```

Comment:
plain LIST_FOREACH should suffice for a loop that doesn't actually remove entries

Reply:
indeed